src/main/java/com/example/spring/service/cart/CartServiceImpl.java

```diff
@@ -0,0 +1,78 @@
+package com.example.spring.service.cart;
+
+import com.example.spring.dto.cartitem.CartItemDto;
+import com.example.spring.dto.cartitem.CartItemRequestDto;
+import com.example.spring.dto.cartitem.UpdateCartItemRequestDto;
+import com.example.spring.dto.shoppingcart.ShoppingCartDto;
+import com.example.spring.exception.EntityNotFoundException;
+import com.example.spring.mapper.CartItemMapper;
+import com.example.spring.mapper.ShoppingCartMapper;
+import com.example.spring.model.CartItem;
+import com.example.spring.model.ShoppingCart;
+import com.example.spring.repository.cartitem.CartItemRepository;
+import com.example.spring.repository.shoppingcart.ShoppingCartRepository;
+import jakarta.transaction.Transactional;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class CartServiceImpl implements CartService {
+    private final CartItemMapper cartItemMapper;
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final ShoppingCartMapper shoppingCartMapper;
+    private final CartItemRepository cartItemRepository;
+
+    @Override
+    @Transactional
+    public ShoppingCartDto save(String email, CartItemRequestDto cartItemRequestDto) {
+        CartItem cartItem = cartItemMapper.toModel(cartItemRequestDto);

```

Comment:
Find Book in DB first
Check if Shop Cart already has the same book, if so -> increase quantity
if no -> add to Shop Cart